File: Csharp/LeetCoding/UniqueMorseCodeWords.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LeetCode
{
    public partial class LeetCodeLibrary
    {
        #region UniqueMorseRepresentations Testing
        public static void UniqueMorseRepresentationsTest()
        {
            string[] words = { "gin", "zen", "gig", "msg" };
            Console.WriteLine(2 == UniqueMorseRepresentations(words));
            string[] words2 = { "hello", "world", "a", "bee"};
            Console.WriteLine(4 == UniqueMorseRepresentations(words2));
        }
        #endregion
        #region UniqueMorseRepresentations
        public static int UniqueMorseRepresentations(string[] words)
        {
            var morseWordSet = new HashSet<string>();
            string[] morseAlphabet = { ".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.." };
            foreach (string word in words)
            {
                StringBuilder morse_word = new StringBuilder();
                foreach (char letter in word)
                    morse_word.Append(morseAlphabet[((int)letter) - 97]);
                morseWordSet.Add(morse_word.ToString());
            }
            return morseWordSet.Count;
        }
        #endregion
    }
    
}

File: Python/uniqueMorseCodeWords.py
from mock import MagicMock
def uniqueMorseRepresentations( words ):
    """
    :type words: List[str]
    :rtype: int
    """
    morse_words = set() #set because they want distinct count
    morse_alphabet = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
    for word in words:
        morse_word=""
        for letter in word:
            morse_word+=morse_alphabet[ord(letter)-97]
        morse_words.add(morse_word)
    return len(morse_words)


def uniqueMorseRepresentationsTesting():
    words = ["gin", "zen", "gig", "msg"]
    print(2==uniqueMorseRepresentations(words))
    words = ["hello", "world", "a", "bee"]
    print(4==uniqueMorseRepresentations(words))

'''Notes
    The length of words will be at most 100.
    Each words[i] will have length in range [1, 12].
    words[i] will only consist of lowercase letters.
'''
uniqueMorseRepresentationsTesting()